### DISCUSSION TITLE
Help with ace interation

### DISCUSSION THREAD

**Discussion by RipperRabbit** [acemod/ACE3#8667] [Q&A]
Hi

I have been trying to make some colored panels that you can put on the ground and pick back up to mark HLS's that's an inventory item much like the tripod and the spotting scope. I have been able to get it to show in the interaction menu, but nothing happens when you select it, so I'm not sure what is going on or if I have missed anything.

**Comment by severgun** [comment]
Without code example it is impossible to tell.
Interaction menu entry have `statement` section. Code in this section will be executed when selected.
So if there is no code that will take or spawn object nothing will happen.

**Reply by RipperRabbit** [reply]
Okay so this is what I have so far but still can not work it out trying to get it so if you have the HLS panel item you can place on the ground.

#define _ARMA_

class CfgPatches
{
	class 16aa_HLS_Panels_c
	{
		units[] = {};
		weapons[] = {};
		requiredVersion = 0.1;
		requiredAddons[] = {"16aa_HLS_Panels"};
	};
};

class CfgWeapons
{
    class ACE_ItemCore;
    class CBA_MiscItem_ItemInfo;
    class 16aa_Item_HSL_Panel_Y: ACE_ItemCore 
    {
        scope = 2;
        displayName = "HLS Panel (Yellow)";
        descriptionShort = "Yellow HLS Marking Panel used for marking a HLS";
        model = "\lsr_rework\addons\hls_pannels\16aa_object_signs_HLS_panel_yellow_object.p3d";
        picture = "\lsr_rework\addons\hls_pannels\UI\panel_yellow.paa";
        class ItemInfo: CBA_MiscItem_ItemInfo 
        {
        mass = 0.1;
        };
    };
};

class CfgVehicles
{
    class Man;
    class CAManBase: Man
    {
        class ACE_SelfActions
        {
            class ACE_Equipment
            {
                // ACE Placeing 
                class 16aa_Pickup_HLS_Panel_Yellow 
                {
                    displayName = "Place HLS Panel (Yellow)";
                    condition = QUOTE([ARR_2(_player,'16aa_Item_HSL_Panel_Y')] call ace_common_fnc_hasItem);
                    statement = QUOTE([ARR_4(_player,'16aa_Item_HSL_Panel_Y','16aa_Object_HLS_Panel_Red',1)] call EFUNC(placeTimer));
                    showDisabled = 0;
                    priority = 2;
                    icon = "\lsr_rework\addons\hls_pannels\UI\panel_yellow.paa";
                };

            };
        };   
    };

    class ReammoBox_F;
    class 16aa_Object_HLS_Panels_Base: ReammoBox_F 
    {
        scope = 0;
        //editorCategory = "16aa_Eden_Category";
        //editorSubcategory = "16aa_Eden_Subcategory_Object_Signs";
        icon = "iconObject_10x1";
        mapsize = 10;
        damageResistance = 10;
        armor = 10000;
    };

    class 16aa_Object_HLS_Panels_Y: 16aa_Object_HLS_Panels_Base 
    {
        scope = 2;
        displayName = "HLS Panel (Yellow)";
        model = "\lsr_rework\addons\hls_pannels\16aa_object_signs_HLS_panel_yellow_object.p3d";
        ace_dragging_canDrag = 1;
        ace_dragging_canCarry = 1;
        class ACE_Actions 
        {
            class ACE_MainActions 
            {
                selection = "interact";
                distance = 5;
                condition = "true";
                displayName = "Interact";
                class 16aa_Pickup_HLS_Panel_Yellow 
                {
                    selection = "";
                    displayName = "Pick up HLS Panel";
                    distance = 5;
                    condition = "true";
                    statement = QUOTE([ARR_4(_target,_player,'16aa_Item_HSL_Panel_Y',1)] call EFUNC(pickupTimer));
                    showDisabled = 0;
                    exceptions[] = {};
                    priority = 4;
                };
            };
        };
    };

};
